src/conversions/chrono.rs

```diff
@@ -0,0 +1,848 @@
+    fn extract(ob: &PyAny) -> PyResult<NaiveDate> {
+        let date = <PyDate as PyTryFrom>::try_from(ob)?;
+        Ok(NaiveDate::from_ymd(
+            date.get_year(),
+            date.get_month() as u32,
+            date.get_day() as u32,
+        ))
+    }
+}
+
+impl ToPyObject for NaiveTime {
+    fn to_object(&self, py: Python<'_>) -> PyObject {
+        let h = self.hour() as u8;
+        let m = self.minute() as u8;
+        let s = self.second() as u8;
+        let ns = self.nanosecond();
+        let (ms, fold) = match ns.checked_sub(1_000_000_000) {
+            Some(ns) => (ns / 1000, true),
+            None => (ns / 1000, false),
+        };
+        let time =
+            PyTime::new_with_fold(py, h, m, s, ms, None, fold).expect("Failed to construct time");
+        time.into()
+    }
+}
+
+impl IntoPy<PyObject> for NaiveTime {
+    fn into_py(self, py: Python<'_>) -> PyObject {
+        ToPyObject::to_object(&self, py)
+    }
+}
+
+impl FromPyObject<'_> for NaiveTime {
+    fn extract(ob: &PyAny) -> PyResult<NaiveTime> {
+        let time = <PyTime as PyTryFrom>::try_from(ob)?;
+        let ms = time.get_fold() as u32 * 1_000_000 + time.get_microsecond();
+        let h = time.get_hour() as u32;
+        let m = time.get_minute() as u32;
+        let s = time.get_second() as u32;
+        Ok(NaiveTime::from_hms_micro(h, m, s, ms))
+    }
+}
+
+impl<Tz: TimeZone> ToPyObject for DateTime<Tz> {
+    fn to_object(&self, py: Python<'_>) -> PyObject {
+        let date = self.naive_utc().date();
+        let time = self.naive_utc().time();
+        let yy = date.year();
+        let mm = date.month() as u8;
+        let dd = date.day() as u8;
+        let h = time.hour() as u8;
+        let m = time.minute() as u8;
+        let s = time.second() as u8;
+        let ns = time.nanosecond();
+        let (ms, fold) = match ns.checked_sub(1_000_000_000) {
+            Some(ns) => (ns / 1000, true),
+            None => (ns / 1000, false),
+        };
+        let tz = self.offset().fix().to_object(py);
+        let tz = tz.cast_as(py).unwrap();
+        let datetime = PyDateTime::new_with_fold(py, yy, mm, dd, h, m, s, ms, Some(tz), fold)
+            .expect("Failed to construct datetime");
+        datetime.into()
+    }
+}
+
+impl<Tz: TimeZone> IntoPy<PyObject> for DateTime<Tz> {
+    fn into_py(self, py: Python<'_>) -> PyObject {
+        ToPyObject::to_object(&self, py)
+    }
+}
+
+impl FromPyObject<'_> for DateTime<FixedOffset> {
+    fn extract(ob: &PyAny) -> PyResult<DateTime<FixedOffset>> {
+        let dt = <PyDateTime as PyTryFrom>::try_from(ob)?;
+        let ms = dt.get_fold() as u32 * 1_000_000 + dt.get_microsecond();
+        let h = dt.get_hour().into();
+        let m = dt.get_minute().into();
+        let s = dt.get_second().into();
+        let tz = if let Some(tzinfo) = dt.get_tzinfo() {
+            tzinfo.extract()?
+        } else {
+            return Err(PyTypeError::new_err("Not datetime.timezone.tzinfo"));
```

Comment:
I think it should just be `datetime.tzinfo`? `datetime.timezone` being the fixed offset representation.

```suggestion
            return Err(PyTypeError::new_err("Not datetime.tzinfo"));
```